server/controllers/match/placementRequests/occupancyViewController.ts

```diff
@@ -29,32 +43,68 @@ export default class {
 
   view(): TypedRequestHandler<Request> {
     return async (req: NewRequest, res: Response) => {
-      const { startDate, durationDays, premisesName, premisesId, apType } = req.query
-      const dates = placementDates(startDate, durationDays)
-      const [placementRequest, capacity] = await Promise.all([
-        this.placementRequestService.getPlacementRequest(req.user.token, req.params.id),
-        this.premisesService.getCapacity(req.user.token, premisesId, dates.startDate, dates.endDate),
-      ])
-      const essentialCharacteristics = filterOutAPTypes(placementRequest.essentialCriteria)
-      const matchingDetailsSummaryList = occupancyViewSummaryListForMatchingDetails(
-        capacity.premise.bedCount,
-        dates,
-        placementRequest,
-        essentialCharacteristics,
-      )
-      const occupancySummaryHtml = occupancySummary(capacity)
-      const calendar = occupancyCalendar(capacity)
+      const { token } = req.user
+      const { id, premisesId } = req.params
+      const { apType, criteria, ...filterUserInput } = req.query
       const { errors, errorSummary, userInput } = fetchErrorsAndUserInput(req)
+
+      const placementRequest = await this.placementRequestService.getPlacementRequest(token, id)
+      const premises = await this.premisesService.find(token, premisesId)
+
+      let criteriaAsArray: Array<Cas1SpaceBookingCharacteristic>
+      if (criteria) {
+        criteriaAsArray = Array.isArray(criteria) ? criteria : [criteria]
+      }
+      const durationDays = Number(filterUserInput.durationDays) || placementRequest.duration
+
+      let startDate: string
+      let dateFieldValues: ObjectWithDateParts<'startDate'>
+
+      if (filterUserInput['startDate-day'] || filterUserInput['startDate-month'] || filterUserInput['startDate-year']) {
+        dateFieldValues = filterUserInput
+
+        if (dateAndTimeInputsAreValidDates(filterUserInput, 'startDate')) {
+          startDate = DateFormats.dateAndTimeInputsToIsoString(filterUserInput, 'startDate').startDate
+        } else {
+          const dateError = { startDate: 'Enter a valid date' }
+          Object.assign(errors, generateErrorMessages(dateError))
+          errorSummary.push(generateErrorSummary(dateError)[0])
+        }
+      } else {
+        startDate = placementRequest.expectedArrival
+        dateFieldValues = DateFormats.isoDateToDateInputs(startDate, 'startDate')
+      }
```

Comment:
I'm not sure it would be clearer, as this logic touches a lot of different symbols -- it would need to be aware of `filterUserInput`, `errors`, `errorSummary`, `placementRequest`, and would also end up mutating both `errors` and `errorSummary` which are shared with other bits of the logic, as well as returning both `startDate` and `dateFieldValues`... It's not logic that is reused anywhere either!

Reply:
not for reuse necessarily - more so that the function does not do so much you cannot see the wood from the trees etc. I guess it's only on the borders for now and we have almost finished this page.
For interest I was thinking something along these lines:
```
const { startDate, durationDays, dateFieldValues } = this.getDatesAndDurationFromFilters(
        filterUserInput,
        placementRequest,
        errors,
        errorSummary,
)
```